src/main/java/team/asd/util/MessageBuilder.java

```diff
@@ -0,0 +1,94 @@
+package team.asd.util;
+
+import org.apache.commons.lang3.StringUtils;
+import team.asd.entities.BaseLocation;
+import team.asd.entities.IsLocation;
+import team.asd.entities.IsLocationSearchRequest;
+
+import java.lang.reflect.Field;
+
+public class MessageBuilder {
+	private final StringBuilder messageBuilder = new StringBuilder();
+	private String exceptionInformation = null;
+
+	public <T> MessageBuilder add(T value, String fieldName) {
+		if (fieldName == null) {
+			return add(value);
+		}
+		addSpaceIfNeeded();
+		messageBuilder.append(fieldName).append(": ");
+		if (value == null) {
+			messageBuilder.append("null");
+		} else {
+			if (value instanceof String) {
+				messageBuilder.append('"').append(value).append('"');
+			} else {
+				messageBuilder.append(value);
+			}
+		}
+		return this;
+	}
+
+	public <T> MessageBuilder add(T value) {
+		if (value instanceof String) {
+			if (StringUtils.isBlank((String) value)) {
+				return this;
+			}
```

Comment:
These nested if statements could be combined using `&&`.
E.g.:
`if (value instanceof String && StringUtils.isBlank((String) value))`